ios/browser/api/ads/brave_ads.mm

```diff
@@ -1587,18 +1587,25 @@ - (void)triggerNewTabPageAdEvent:(NSString*)wallpaperId
       base::BindOnce(completion));
 }
 
-- (nullable NotificationAdIOS*)maybeGetNotificationAd:(NSString*)identifier {
+- (void)maybeGetNotificationAd:(NSString*)identifier
+                    completion:(void (^)(NotificationAdIOS*))completion {
   if (![self isServiceRunning]) {
-    return nil;
+    completion(nil);
+    return;
   }
 
-  const std::optional<brave_ads::NotificationAdInfo> ad =
-      ads->MaybeGetNotificationAd(identifier.UTF8String);
-  if (!ad) {
-    return nil;
-  }
+  ads->MaybeGetNotificationAd(
+      identifier.UTF8String,
```

Comment:
Since we're making changes here can we replace this old conversion with `base::SysNSStringToUTF8`

Reply:
Changed all `.UTF8String` to `base::SysNSStringToUTF8`.